pkg/sql/conn_executor_exec.go

```diff
@@ -970,7 +970,7 @@ func (ex *connExecutor) dispatchToExecutionEngine(
 
 	// Certain statements want their results to go to the client
 	// directly. Configure this here.
-	if planner.curPlan.avoidBuffering {
+	if planner.curPlan.avoidBuffering || ex.sessionData().AvoidBuffering {
```

Comment:
I don't really have much expertise in the sql session / serialize session / exec stuff -- my weak preference would be to split out the avoid buffering bits to their own PR that someone on SQL could stamp too, but it's a weak preference.

Reply:
Yeah; I spoke w/ Yahor on this.  I just want to have this together w/ the use case for this setting.
@yuzefovich do you mind taking a peak at the AvoidBuffering bit?